src/shims/time.rs

```diff
@@ -64,7 +64,6 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     ) -> InterpResult<'tcx, i32> {
         let this = self.eval_context_mut();
 
-        this.assert_target_os("macos", "gettimeofday");
```

Comment:
This should probably be `assert_target_os_is_unix`.
Also on its own this will do nothing, the assertion here is just a double-check. The only place this function is called is for macOS.

Reply:
Sounds good, changed.

Reply:
As I said, on its own this will do nothing.

You need to actually wire this up so that it is called on Linux as well. Please run tests locally before pushing to this PR, it will clearly tell you that `gettimeofday` is still not supported when you run it on a Linux target.

You will have to move this
https://github.com/rust-lang/miri/blob/328a8c7a1e9dbf3b86e867982f26ad3df5d38436/src/shims/unix/macos/foreign_items.rs#L81
to somewhere suitable in https://github.com/rust-lang/miri/blob/master/src/shims/unix/foreign_items.rs.

Reply:
Ah, there it is. :)